src/cpu/aarch64/jit_aarch64_sve_512_conv_kernel.cpp

```diff
@@ -0,0 +1,4417 @@
+        return status::unimplemented;
+    jcp.ic_tail = is_data_layout_nxc ? jcp.ic % jcp.simd_w : 0;
+    jcp.oc_tail = is_data_layout_nxc ? jcp.oc % jcp.simd_w : 0;
+
+    format_tag_t dat_tag, wei_tag;
+    const auto nxc_tag = pick(ndims - 3, nwc, nhwc, ndhwc);
+
+    if (jcp.simd_w == 8) {
+        assert(with_groups);
+        dat_tag = is_data_layout_nxc ? nxc_tag
+                                     : pick(ndims - 3, nCw8c, nChw8c, nCdhw8c);
+        wei_tag = pick(ndims - 3, gOIw8o8i, gOIhw8o8i, gOIdhw8o8i);
+    } else if (jcp.simd_w == 4) {
+        assert(with_groups);
+        dat_tag = is_data_layout_nxc ? nxc_tag
+                                     : pick(ndims - 3, nCw4c, nChw4c, nCdhw4c);
+        wei_tag = pick(ndims - 3, gOIw4o4i, gOIhw4o4i, gOIdhw4o4i);
+    } else {
+        dat_tag = is_data_layout_nxc
+                ? pick(ndims - 3, nwc, nhwc, ndhwc)
+                : pick(ndims - 3, nCw16c, nChw16c, nCdhw16c);
+        wei_tag = pick(2 * ndims - 6 + with_groups, OIw16o16i, gOIw16o16i,
+                OIhw16o16i, gOIhw16o16i, OIdhw16o16i, gOIdhw16o16i);
+    }
+
+    if (diff_src_md.format_kind == format_kind::any) {
+        CHECK(memory_desc_init_by_tag(diff_src_md, dat_tag));
+    } else if (curr_src_tag != dat_tag)
+        return status::unimplemented;
+    jcp.src_tag = dat_tag;
+
+    if (diff_dst_md.format_kind == format_kind::any) {
+        CHECK(memory_desc_init_by_tag(diff_dst_md, dat_tag));
+    } else if (curr_dst_tag != dat_tag)
+        return status::unimplemented;
+    jcp.dst_tag = dat_tag;
+
+    if (init_tag(jcp.wei_tag, weights_md, weights_d, wei_tag)
+            != status::success)
+        return status::unimplemented;
+
+    jcp.nb_ic = div_up(jcp.ic, jcp.ic_block);
+    jcp.nb_oc = div_up(jcp.oc, jcp.oc_block);
+
+    jcp.ur_w = jcp.stride_w;
+
+    int regs = 24;
+    if (jcp.iw <= regs)
+        jcp.ur_w = jcp.iw;
+    else {
+        for (int ur_w = regs; ur_w > 0; --ur_w)
+            if (ur_w % jcp.stride_w == 0) {
+                jcp.ur_w = ur_w;
+                break;
+            }
+    }
+    int l_overflow = nstl::max(
+            0, ((jcp.kw - 1) * (jcp.dilate_w + 1) - jcp.l_pad) / jcp.stride_w);
+    int r_overflow_no_tail = nstl::max(0,
+            ((jcp.kw - 1) * (jcp.dilate_w + 1)
+                    - nstl::max(0, jcp.r_pad + jcp.iw % jcp.ur_w))
+                    / jcp.stride_w);
+    int n_oi = jcp.iw / jcp.ur_w;
+    if (r_overflow_no_tail > 0) n_oi--;
+
+    if (mayiuse(sve_512) && diff_dst_d.data_type() == data_type::f32
+            && weights_d.data_type() == data_type::f32
+            && diff_src_d.data_type() == data_type::f32) {
+        jcp.ver = ver_fma;
+        jcp.typesize_in = typesize;
+        jcp.typesize_out = typesize;
+    } else {
+        return status::unimplemented;
+    }
+
+    if (!utils::everyone_is(0, jcp.dilate_d, jcp.dilate_h, jcp.dilate_w)
+            && jcp.ver != ver_fma)
+        return status::unimplemented;
```

Comment:
I don't see this as required since there are no other versions (`jcp.ver`) for `ver_fma`.

Reply:
Thx!
I'll delete this IF-block.